carma_ros2_utils/src/component_manager.cpp

```diff
@@ -281,16 +281,38 @@ ComponentManager::on_load_node(
             sev = RCUTILS_LOG_SEVERITY_WARN;
           }
           // Set the log level
-          auto result = rcutils_logging_set_logger_level(node_wrappers_[node_id].get_node_base_interface()->get_name(), sev);
-          
+
+          // Modify the node's fully qualified name to match the logger's name with namespaces
+          const char* name =  node_wrappers_[node_id].get_node_base_interface()->get_fully_qualified_name();
+
+          char modified_name[strlen(name) + 1]; // +1 for the null terminator
+
+          // Iterate through the original string, replacing '/' with '.' and skipping the first character if it's '/'
+          int j = 0;
+          for (int i = (name[0] == '/' ? 1 : 0); name[i] != '\0'; ++i) {
+              modified_name[j++] = (name[i] == '/' ? '.' : name[i]);
+          }
+          modified_name[j] = '\0'; // Null-terminate the modified string
+
+          auto result = rcutils_logging_set_logger_level(modified_name, sev);
+
           if (result != RCUTILS_RET_OK) {
-            RCLCPP_ERROR(get_logger(), "FAILED to set log level when provided with --log-level argument");
+            RCLCPP_ERROR_STREAM(get_logger(), "FAILED to set log level using the node's name with namespaces when provided with --log-level argument, name: " << std::string(modified_name));
```

Comment:
With the above change, `std::string(modified_name)` can be replaced with `name`.